### ISSUE TITLE
test sourmash with python 3.10?

### ISSUE THREAD

**Issue by ctb** [sourmash-bio/sourmash#1704]
per https://twitter.com/brettsky/status/1423360362754232320

**Comment by luizirber** [comment]
ooh, I didn't know about `--build-number`, that is pretty useful...

That said, the sourmash wheel works with 3.10 (or whatever python version) as long as `cffi` is installable, so while it is nice to test in `3.10` we don't need to build new wheels.

we could also run tests against `python-dev` in CI (scheduled job, daily), but that can take a LONG time because as Brett mentioned most (all?) packages don't publish wheels before the actual Python release, so it needs to recompile all our dependencies...

**Comment by ctb** [comment]
I created a new conda environment with python 3.10, and then tried to pip install sourmash. My laptop fell afoul of an inability to compile scipy, and no scipy version was available for python 3.10 via conda. I'll leave this open for later, I guess ;).

**Comment by ctb** [comment]
installing sourmash via pip in a python=3.10 environment worked:
```
mamba create -n py10 -y python=3.10
conda activate py10
pip install sourmash
sourmash
```

as did installing it in dev mode and running tests:
```
pip install -e '.[all]'
pytest tests
```

🎉 

and, somewhere along the way, python3.10 tests got added to github actions. Closing!! 🎉
